pkg/adapter/adapter.go

```diff
@@ -104,10 +105,28 @@ func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, httpMe
 	}
 }
 
+func VhostHandler(Brokers string, Vhost string) string {
```

Comment:
If we don't expect to need to call this from any other place, maybe change this to:
vhostHandler

In go, a function starting with lowercase is not visible outside the package, so limiting to smallest scope possible is typically what you want.